sdk/storage/azure-storage-blobs/test/blob_container_client_test.cpp

```diff
@@ -1048,4 +1048,39 @@ namespace Azure { namespace Storage { namespace Test {
     }
   }
 
+  TEST_F(BlobContainerClientTest, SpecialBlobName)
```

Comment:
Great test. Does this cover all the Url encoding cases or are their other APIs that could benefit from similar tests?